### ISSUE TITLE
Dependi fails to identify invalid versions in certain circumstances

### ISSUE THREAD

**Issue by donottellmetonottellyou** [filllabs/dependi#112]
**Describe the bug**
If the `godot` crate is set to the invalid version of `0.0.1` (with a lock file having the correct version), it will show a green check next to the version even though it may not even exist.

**To Reproduce**
1. Make sure dependi has the lock file feature enabled
2. Create a project depending on `godot` like this: `godot = "0.1.3"`
3. Run `cargo build` or let dependi generate a lock file 
4. Change the version of the crate to an invalid previous version `godot = "0.0.1"`
5. Note the green check, rustup has crashed
6. Cargo build:

```
error: failed to select a version for the requirement `godot = "^0.0.1"`
candidate versions found which didn't match: 0.1.3, 0.1.2, 0.1.1
```

**Expected behavior**
Dependi should detect when an invalid version exists in the `Cargo.toml`, even when `Cargo.lock` is up-to-date.

**Screenshots**

![image](https://github.com/user-attachments/assets/1423e51f-7640-4a69-8efb-f8044b496b5e)

**Desktop (please complete the following information):**
 - OS: NixOS Linux
 - Version: 24.05

**Additional context**
Dependi `0.7.8`
Related to #15

**Comment by MOzanGul** [comment]
Hi @donottellmetonottellyou, 
We fixed this issue on our newest version 0.7.11. 
Thank you for contributing!

**Comment by MOzanGul** [comment]
@donottellmetonottellyou 
We'd be really happy if you could review Dependi, here is our [reviews page](https://marketplace.visualstudio.com/items?itemName=fill-labs.dependi&ssr=false#review-details).
